RealmSwift/Tests/RealmTests.swift

```diff
@@ -0,0 +1,462 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2015 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import XCTest
+import RealmSwift
+import Foundation
+
+class RealmTests: TestCase {
+    override func setUp() {
+        super.setUp()
+        realmWithTestPath().write {
+            SwiftStringObject.createInRealm(self.realmWithTestPath(), withObject: ["1"])
+            SwiftStringObject.createInRealm(self.realmWithTestPath(), withObject: ["2"])
+            SwiftStringObject.createInRealm(self.realmWithTestPath(), withObject: ["3"])
+        }
+
+        Realm().write {
+            SwiftIntObject.createInRealm(Realm(), withObject: [100])
+            SwiftIntObject.createInRealm(Realm(), withObject: [200])
+            SwiftIntObject.createInRealm(Realm(), withObject: [300])
+        }
+    }
+
+    func testPath() {
+        XCTAssertEqual(Realm.defaultPath, Realm().path)
+        XCTAssertEqual(testRealmPath(), realmWithTestPath().path)
+    }
+
+    func testReadOnly() {
+        autoreleasepool {
+            XCTAssertEqual(Realm().readOnly, false)
+        }
+        let readOnlyRealm = Realm(path: Realm.defaultPath, readOnly: true, error: nil)!
+        XCTAssertEqual(true, readOnlyRealm.readOnly)
+        XCTAssertEqual(3, readOnlyRealm.objects(SwiftIntObject).count)
+
+        assertThrows(Realm(), "Realm has different readOnly settings")
+    }
+
+    func testSchema() {
+        let schema = Realm().schema
+        XCTAssert(schema as AnyObject is Schema)
+        XCTAssertEqual(1, schema.objectSchema.filter({ $0.className == "SwiftStringObject" }).count)
+    }
+
+    func testDefaultPath() {
+        let defaultPath =  Realm().path
+        XCTAssertEqual(Realm.defaultPath, defaultPath)
+
+        let newPath = defaultPath.stringByAppendingPathExtension("new")!
+        Realm.defaultPath = newPath
+        XCTAssertEqual(Realm.defaultPath, newPath)
+        XCTAssertEqual(Realm().path, Realm.defaultPath)
+    }
+
+    func testInit() {
+        XCTAssertEqual(Realm().path, Realm.defaultPath)
+        XCTAssertEqual(Realm(path: testRealmPath()).path, testRealmPath())
+        assertThrows(Realm(path: ""))
+    }
+
+    func testInitFailable() {
+        var error: NSError?
+        autoreleasepool({
```

Comment:
I think we've been consistent about dropping the parens for `autoreleasepool`. There's also a few `realm.write`s below with extra parens.